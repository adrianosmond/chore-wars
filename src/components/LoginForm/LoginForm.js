import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { auth } from 'utils/database';
import * as routes from 'constants/routes';

import './LoginForm.css';

const INITIAL_STATE = {
  email: '',
  password: '',
  error: null,
};

class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = INITIAL_STATE;
  }

  onSubmit = (event) => {
    event.preventDefault();
    const { email, password } = this.state;
    const { loggedIn } = this.props;
    auth.signInWithEmailAndPassword(email, password)
      .then(() => {
        this.setState({ ...INITIAL_STATE });
        loggedIn();
      })
      .catch((error) => {
        this.setState({
          error: error.message,
        });
      });
  }

  render() {
    const { error, email, password } = this.state;
    return (
      <div className="login">
        <div className="login__widget">
          <h1 className="login__title">Chore Wars</h1>
          <p>{error || 'Please enter your email and password to log in'}</p>
          <form onSubmit={this.onSubmit} className="form form--contained">
            <label className="form__label form__label--small" htmlFor="email">Email</label>
            <input
              className="form__input"
              id="email"
              placeholder="Email"
              type="email"
              onChange={event => this.setState({ email: event.target.value })}
              value={email}
            />
            <label className="form__label form__label--small" htmlFor="password">Password</label>
            <input
              className="form__input"
              id="password"
              placeholder="Password"
              type="password"
              onChange={event => this.setState({ password: event.target.value })}
              value={password}
            />
            <button type="submit" className="form__button">Login</button>
          </form>
          <p className="login__link">
            <Link to={routes.SIGN_UP}>Create an account</Link>
          </p>
        </div>
      </div>
    );
  }
}

export default LoginForm;
