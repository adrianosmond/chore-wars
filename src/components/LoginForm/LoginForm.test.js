import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './LoginForm';

describe('LoginForm', () => {
  it('Renders', () => {
    expect(shallow(
      <LoginForm loggedIn={jest.fn()} />,
    )).toMatchSnapshot();
  });
});
