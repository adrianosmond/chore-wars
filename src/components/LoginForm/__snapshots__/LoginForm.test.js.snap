// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Renders 1`] = `
<div
  className="login"
>
  <div
    className="login__widget"
  >
    <h1
      className="login__title"
    >
      Chore Wars
    </h1>
    <p>
      Please enter your email and password to log in
    </p>
    <form
      className="form form--contained"
      onSubmit={[Function]}
    >
      <label
        className="form__label form__label--small"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="form__input"
        id="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
      <label
        className="form__label form__label--small"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form__input"
        id="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <button
        className="form__button"
        type="submit"
      >
        Login
      </button>
    </form>
    <p
      className="login__link"
    >
      <Link
        replace={false}
        to="/sign-up"
      >
        Create an account
      </Link>
    </p>
  </div>
</div>
`;
